### DISCUSSION TITLE
Celery+Redis Queue Priority

### DISCUSSION THREAD

**Discussion by harshraj22** [celery/celery#8226] [Q&A]
I have 3 queues with priority Low, Medium, and High. With the default options shown in the tutorials, the tasks in Low priority queue will not be executed as long as there is any pending task in Medium/ High priority queue. So, if tasks are continiously being added to Medium and High priority queue such that they both are never empty simultaneously, the low priority tasks will never be executed. Is there any way that sometimes I can allow some Low priority tasks to be executed as well while Hight/ Medium priority tasks are still in the queue ?

This will ensure that Low priority tasks don't have to wait tooo long/ infinitely.

**Comment by pfk-beta** [comment]
Hi,
I'm not an celery expert, but I would do it in this way:
I would monitor tasks with low priority, then I would change priority of old and low priority to medium priority. I would do it programatically, bacause I don't recall any mechanism inside celery.

**Reply by harshraj22** [reply]
Yeah. using such manual tricks are always an option, but I was looking for some solution using celery configurations itself (if there's one)
